src/evict.c

```diff
@@ -342,7 +342,7 @@ size_t freeMemoryGetNotCountedMemory(void) {
         }
     }
     if (server.aof_state != AOF_OFF) {
-        overhead += sdsalloc(server.aof_buf)+aofRewriteBufferSize();
+        overhead += sdsZmallocSize(server.aof_buf)+aofRewriteBufferSize();
```

Comment:
i'm a bit concerned about the performance implications of calling mallc_size here (per command).
maybe we should use sdsAllocSize instead (which is similar to sdsalloc except that it also counts the sds header and null terminator)
i.e. the difference between these two is that one (current one that is wrongly used) is about the size of data we can store in the sds, and the other one i'm suggesting is about the memory it uses.

note that after #7875 there is actually no difference between what i'm suggesting and the current code in the PR (except that one is cleaner, but also possibly slower).
this also means that the current code before this fix, is now (after #7875) not that bad...

Reply:
Oh, I miss that, redis do call this per command when memory is almost fully, yes, after #7875, there is no difference between `sdsZmallocSize` and `sdsAllocSize` if allocator has alloc size function.

Reply:
ok. so let's take sdsAllocSize.. a bit less clear, but should be faster in theory.